File: app/src/main/java/com/ufkoku/demo_app/DemoApp.java
package com.ufkoku.demo_app;

import android.app.Application;
import android.os.StrictMode;

public class DemoApp extends Application {

    @Override
    public void onCreate() {
        StrictMode.setThreadPolicy(
                new StrictMode.ThreadPolicy.Builder()
                        .detectAll()
                        .penaltyLog()
                        .build()
        );
        StrictMode.setVmPolicy(
                new StrictMode.VmPolicy.Builder()
                        .detectAll()
                        .penaltyLog()
                        .build()
        );

        super.onCreate();
    }

}

File: app/src/main/kotlin/com/ufkoku/demo_app/DemoApp.kt
package com.ufkoku.demo_app

import android.app.Application
import android.os.StrictMode
import com.squareup.leakcanary.LeakCanary

class DemoApp : Application() {

    override fun onCreate() {
        if (LeakCanary.isInAnalyzerProcess(this)) {
            return
        }

        super.onCreate()

        StrictMode.setThreadPolicy(
                StrictMode.ThreadPolicy.Builder()
                        .detectAll()
                        .penaltyLog()
                        .build())

        StrictMode.setVmPolicy(
                StrictMode.VmPolicy.Builder()
                        .detectAll()
                        .penaltyLog()
                        .build())

        LeakCanary.install(this)
    }

}
